### PR TITLE
[PIR] Change "new_ir" to "pir"

### PR DESCRIPTION
**Pull request by yuanlehome**
### PR types

Others
### PR changes

Others
### Description

涉及到FLAGS、变量、函数、类、文件、文件夹等等名称中包含"new ir"字样的替换，规则如下：
- "new_ir", "newir ---> "pir"
- "NEW_IR", "NEWIR", "NewIR", "NewIr" ---> "Pir"
### Others
Pcard-71500

### CODE REVIEW

File: paddle/fluid/eager/to_static/run_program_op_func.h [hunk 1]
```diff
@@ -232,7 +232,7 @@ inline void newir_run_program_ad_func(
   auto output_size =
       PADDLE_GET_CONST(std::vector<::pir::Value>, attrs.at("fo")).size();
   auto middles = std::vector<paddle::Tensor*>();
-  std::shared_ptr<NewIRGradNodeRunProgram> grad_node;
+  std::shared_ptr<PIRGradNodeRunProgram> grad_node;
```
Comment (zhangbo9674) [inline paddle/fluid/eager/to_static/run_program_op_func.h]:
```suggestion
  std::shared_ptr<PirGradNodeRunProgram> grad_node;
```

File: paddle/fluid/eager/to_static/run_program_op_func.h [hunk 2]
```diff
@@ -266,7 +266,7 @@ inline void newir_run_program_ad_func(
 
   // Call forward function
   // if require_any_grad is False, don't save any middle vars.
-  NewIRRunProgramAPI(
+  PIRRunProgramAPI(
```
Comment (zhangbo9674) [inline paddle/fluid/eager/to_static/run_program_op_func.h]:
```suggestion
  PirRunProgramAPI(
```

**Review by XieYunshen — APPROVED** [review]
LGTM

**Review by Aurelius84 — APPROVED** [review]
Great work!

**Review by zhangbo9674 — APPROVED** [review]

**Review by zyfncg — APPROVED** [review]
LGTM for setup.py.in

**Review by wanghuancoder — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**